khard/helpers/__init__.py

```diff
@@ -61,6 +62,101 @@ def get_random_uid() -> str:
     return ''.join([random.choice(string.ascii_lowercase + string.digits)
                     for _ in range(36)])
 
+def yaml_clean(value: Union[str, List]) -> Optional[Union[List[str], str]]:
+    """
+    sanitize yaml values according to some simple principles:
+      1. empty values are none, so ruamel does not print an empty list/str
+      2. list with only one item become this item
+      3. multiline strings use the YAML literal style:
+         https://yaml.org/spec/1.2.2/#literal-style
+    
+    :param value: the value to be sanitized
+    :returns: the sanitized value
+    """
+    # special case for empty values
+    if not value:
+        value = None
+    if isinstance(value, list):
+        # special case for single item lists:
+        if len(value) == 1 and isinstance(value[0], str):
+            value = value[0]
+    elif isinstance(value, str):
+        if "\n" in value:
+            value = LiteralScalarString(value)
+
+    return value
+
+
+def yaml_dicts(
+        data: Optional[Dict[str, Any]], 
+        defaults: Union[Dict[str, Any], List[str]] = None
+    ) -> Optional[Dict[str, Any]]:
+    """
+    format a dict according to template, if empty use specified defaults
+
+    :param data: dict of contact data with keys as types and values as values.
+    :param defaults: default dict to be used if data is empty.
+    :returns: dict of types and values.
+    """
+    if not data:
+        if isinstance(defaults, list):
+            return dict((key, None) for key in defaults)
+       
+        return defaults
+
+    data_dict = {}
+    for t, nlist in sorted(data.items(), key=lambda k: k[0].lower()):
```

Comment:
Why t and nlist and not key, value?  What do these names stand for, I don't understand them?